### PR TITLE
MM-30359: Cypress/E2E: Account Settings > Email

### PR DESCRIPTION
**Pull request by agnivade**
This PR automates the following test cases:

MM-T2066

MM-T2067

MM-T2068

MM-T2069

MM-T2070

MM-T2071

MM-T2072

MM-T2073

https://mattermost.atlassian.net/browse/MM-30359

```release-note
NONE
```

### CODE REVIEW

File: e2e/cypress/integration/account_settings/general/email_spec.js [hunk 1]
```diff
@@ -48,4 +59,163 @@ describe('Account Settings -> General -> Email', () => {
...
+    });
+
+
+    // it('MM-T2071 Change email address: update email verification prompt ', () => {
+
+    // });
+
+    // it('MM-T2072', () => {
+
+    // });
```
Comment (agnivade) [inline e2e/cypress/integration/account_settings/general/email_spec.js:168]: These test cases are incorrect or already covered by other tests. I have raised a question in the ~UI Test Automation channel.

File: e2e/cypress/integration/account_settings/general/email_spec.js [hunk 2]
```diff
@@ -48,4 +60,160 @@ describe('Account Settings -> General -> Email', () => {
...
+        cy.get('#saveSetting').click().wait(TIMEOUTS.HALF_SEC);
+
+        // * Check that the correct error message is shown.
+        cy.get('#clientError').should('be.visible').should('have.text', 'The new emails you entered do not match.');
+    });
+
+    // This test is a combination of 4 sub-tests because they are sub-parts of the same test.
+    // Doing them individually would have a dependency with the previous test.
+    // Hence, a combined single test for everything is better.
+    it('MM-T2069/MM-T2070/MM-T2071/MM-T2072 email address verification', () => {
```
Comment (saturninoabril) [inline e2e/cypress/integration/account_settings/general/email_spec.js:110]: Noted. I'll combine the test cases into one, matching the steps here so we can save the result into the TM4J.

Comment (saturninoabril) [reply e2e/cypress/integration/account_settings/general/email_spec.js:110]: Combined into [MM-T2069](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin%3Acom.kanoah.test-manager__main-project-page#!/testCase/MM-T2069). Please update the title. Other test cases were archived.
```suggestion
    it('MM-T2069 Email: Can update email address and verify through email notification, () => {
```

File: e2e/cypress/integration/account_settings/general/email_spec.js [hunk 3]
```diff
@@ -48,4 +60,160 @@ describe('Account Settings -> General -> Email', () => {
...
+
+        // # Type new email
+        cy.get('#primaryEmail').should('be.visible').type(`user-${randomId}@example.com`);
+        cy.get('#confirmEmail').should('be.visible').type(`user-${randomId}@example.com`);
+        cy.get('#currentPassword').should('be.visible').type(testUser.password);
+
+        // # Save the settings
+        cy.get('#saveSetting').click().wait(TIMEOUTS.HALF_SEC);
+
+        // * Verify the announcement bar (MM-2071)
```
Comment (saturninoabril) [inline e2e/cypress/integration/account_settings/general/email_spec.js:124]:
```suggestion
        // * Verify the announcement bar
```

File: e2e/cypress/integration/account_settings/general/email_spec.js [hunk 4]
```diff
@@ -48,4 +60,160 @@ describe('Account Settings -> General -> Email', () => {
...
+        cy.wait(TIMEOUTS.ONE_SEC);
+
+        cy.task('getRecentEmail', {username: testUser.username, mailUrl}).then((response) => {
+            // * Verify the subject
+            expect(response.data.subject).to.equal('[Mattermost] Your email address has changed');
+        });
+
+        cy.toAccountSettingsModal();
+
+        // * Verify new email address (MM-2070)
```
Comment (saturninoabril) [inline e2e/cypress/integration/account_settings/general/email_spec.js:165]:
```suggestion
        // * Verify new email address
```

**Review by saturninoabril — CHANGES_REQUESTED** [review]
Thanks @agnivade! LGTM. Tested and all passed. Left comment to update test title.

**Review by saturninoabril — APPROVED** [review]
Thanks @agnivade!

**Review by isacikgoz — APPROVED** [review]
Looks good 👍

**Pull request merged**